backend/migration/src/m20250111_153247_intial.rs

```diff
@@ -0,0 +1,101 @@
+use sea_orm_migration::{prelude::*, schema::*};
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(Users::Table)
+                    .if_not_exists()
+                    .col(ColumnDef::new(Users::Name).not_null().string().unique_key())
+                    .col(
+                        ColumnDef::new(Users::Email)
+                            .not_null()
+                            .string()
+                            .unique_key()
+                            .primary_key(),
+                    )
+                    .col(
+                        ColumnDef::new(Users::CreatedAt)
+                            .not_null()
+                            .date_time()
+                            .extra("DEFAULT NOW()".to_string()),
+                    )
+                    .col(ColumnDef::new(Users::TeamName).string().null())
+                    .col(
+                        ColumnDef::new(Users::Permissions)
+                            .array(ColumnType::String(StringLen::None))
+                            .null(),
+                    )
+                    .to_owned(),
+            )
+            .await
+            .expect("TODO: panic message");
+        manager
+            .create_table(
+                Table::create()
+                    .table(Teams::Table)
+                    .if_not_exists()
+                    .col(
+                        ColumnDef::new(Teams::Name)
+                            .not_null()
+                            .string()
+                            .unique_key()
+                            .primary_key(),
+                    )
+                    .col(
+                        ColumnDef::new(Teams::CreatedAt)
+                            .not_null()
+                            .date_time()
+                            .extra("DEFAULT NOW()".to_string()),
+                    )
+                    .col(ColumnDef::new(Teams::LeaderName).not_null().string())
```

Comment:
It should be also unique. One user can be leader of only one team. You used also one to one relation